Dockerfile

```diff
@@ -5,21 +5,20 @@ ARG VERSION_RUST="nightly-2022-08-23"
 ARG VERSION_BINARYEN="105-1"
 ARG VERSION_WABT="1.0.27-1"
 
-RUN apt-get update && apt-get install wget -y
-RUN apt-get update && apt-get install python3.10 python-is-python3 -y
-RUN apt-get update && apt-get install build-essential -y
+# Install dependencies (including binaryen and wabt)
+RUN apt-get update && apt-get install -y \
+    wget \ 
+    build-essential \
+    python3.10 python-is-python3 \
+    binaryen=${VERSION_BINARYEN} \
+    wabt=${VERSION_WABT}
 
 # Install rust
 RUN wget -O rustup.sh https://sh.rustup.rs && \
     chmod +x rustup.sh && \
     CARGO_HOME=/rust RUSTUP_HOME=/rust ./rustup.sh --verbose --default-toolchain ${VERSION_RUST} --profile minimal --target wasm32-unknown-unknown -y && \
-    rm rustup.sh
-
-# Install wasm-opt
-RUN apt-get update && apt-get install binaryen=${VERSION_BINARYEN}
-
-# Install wabt
-RUN apt-get update && apt-get install wabt=${VERSION_WABT}
+    rm rustup.sh && \
+    chmod -R 777 /rust
```

Comment:
Every user can read, write and execute `/rust`. In this folder there are (or will be): downloads of cargo, rust binaries, `/cargo-target-dir`.